src/qonnx/core/modelwrapper.py

```diff
@@ -145,6 +146,31 @@ def transform(self, transformation, make_deepcopy=True, cleanup=True):
             (transformed_model, model_was_changed) = transformation.apply(transformed_model)
         if cleanup:
             transformed_model.cleanup()
+
+        if apply_to_subgraphs:
+            for node in transformed_model.model.graph.node:
+                transformed_subgraph_attrs = []
+                for idx, attr in enumerate(node.attribute):
+                    if attr.type == onnx.AttributeProto.GRAPH:
+                        # this is a subgraph, add it to the list
+                        subgraph = ModelWrapper(util.qonnx_make_model(attr.g))
+                        # extract all model metadata from loop model and apply to body
+                        for metadata in transformed_model.model.metadata_props:
+                            subgraph.set_metadata_prop(metadata.key, metadata.value)
+                        # apply the transformation to the subgraph
+                        subgraph = subgraph.transform(transformation, make_deepcopy, cleanup, apply_to_subgraphs)
+                        # copy model metadata from the subgraph to the parent model
+                        for metadata in subgraph.model.metadata_props:
+                            transformed_model.set_metadata_prop(metadata.key, metadata.value)
```

Comment:
same comment as above but other way around - if there are metadata props with same key, child will override parent which sounds undesirable. perhaps we need to discuss exactly how metadata handling needs to be in the presence of subgraphs.